tests/Fixer/Phpdoc/PhpdocTypesOrderFixerTest.php

```diff
@@ -522,6 +522,41 @@ public static function provideFixWithAlphaAlgorithmCases(): iterable
             '<?php /** @return A&B<X|Y|Z>&C&D */',
             '<?php /** @return A&D&B<X|Y|Z>&C */',
         ];
+
+        yield [
+            '<?php /** @param A|(B&C) */',
+            '<?php /** @param (C&B)|A */',
+        ];
+
+        yield [
+            '<?php /** @param A&(B&C) */',
+            '<?php /** @param (C&B)&A */',
+        ];
+
+        yield [
+            '<?php /** @param (A&C)|(B&C)|(C&D) */',
+            '<?php /** @param (C&A)|(C&B)|(C&D) */',
+        ];
+
+        yield [
+            '<?php /** @param (A|C)&(B|C)&(C&D) */',
+            '<?php /** @param (C|A)&(C|B)&(C&D) */',
```

Comment:
In fact, this is not a valid type. DNF types are series of `&` joined with `|`, not the other way around. Look [here](https://phpstan.org/r/55cde7b2-9ec3-4397-bea0-7d07dd9450a1), even though phpDoc allows it, in general it's wrong IMHO. I don't know if we should cover invalid(?) cases 🤔.